backend/layers/common/entities.py

```diff
@@ -98,43 +100,34 @@ def empty():
 
 
 @dataclass
-class CollectionId:
+class EntityId:
```

Comment:
Great idea. I initially went for the same approach but I discarded it (but I can't remember the reason - maybe something with dataclasses?). Anyway if it works this is certainly great.